### PR TITLE
Created clone.md for Java Date clone method

### LINKED ISSUE #2835: [Term Entry] Java Date clone
### Reason/inspiration (optional)

We would like a new term entry under Java [Date](https://www.codecademy.com/resources/docs/java/date) for the Java Date `.clone()` method. 
Provide the entry in a markdown file under `docs/content/java/concepts/date/terms/clone/clone.md`

The entry should
    - Have an introductory paragraph about what the method is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the method in use.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Rayatk**
### Description

- Added `Date.clone()` term entry for Java
- Closes #2835

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/date/terms/clone/clone.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Tags:
+  - 'Date'
+  - 'Methods'
+  - 'Objects'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`Date.clone()`** method returns a copy or duplicate of a specified Date object.
```
Comment (caupolicandiaz) [inline content/java/concepts/date/terms/clone/clone.md:16]:
```suggestion
The **`Date.clone()`** method returns a copy or duplicate of a specified `Date` object.
```

File: content/java/concepts/date/terms/clone/clone.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Syntax
+
+```pseudo
+currentDate.clone()
+```
+
+The `Date.clone()` does not take any parameters.
+
+It returns a copy or duplicate of the `currentDate` when the method is called without an argument. However, if an argument is passed on to the method,
```
Comment (caupolicandiaz) [inline content/java/concepts/date/terms/clone/clone.md:26]:
```suggestion
It returns a copy or duplicate of the `currentDate`, and the returned value is an `Object`, not a `Date`.  
```

File: content/java/concepts/date/terms/clone/clone.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+      Date currentDate = new Date();
+	
+      System.out.println("The original date is " + currentDate);
+      System.out.println("The clone of the date is " + currentDate.clone());
+   }
+}
+```
+
+The output will be:
```
Comment (caupolicandiaz) [inline content/java/concepts/date/terms/clone/clone.md:46]:
```suggestion
The output will be similar to:
```

File: content/java/concepts/date/terms/clone/clone.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+currentDate.clone()
+```
+
+The `Date.clone()` does not take any parameters.
+
+It returns a copy or duplicate of the `currentDate` when the method is called without an argument. However, if an argument is passed on to the method, 
+
+## Example
+
+The following example shows the implementation of `Date.clone()`:
```
Comment (caupolicandiaz) [inline content/java/concepts/date/terms/clone/clone.md:30]:
```suggestion
The following example shows an implementation of `Date.clone()`:
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@Rayatk I've added my initial comments. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by Rayatk** [comment]
@caupolicandiaz I have finished making changes as suggested.

**Comment by Rayatk** [comment]
> @Rayatk I've added my initial comments. Please update the entry, and tag me when you've finished making changes. Thanks

@caupolicandiaz I have finished making changes as suggested.

File: content/java/concepts/date/terms/clone/clone.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+The **`Date.clone()`** method returns a copy or duplicate of a specified `Date` object.
+
+## Syntax
+
+```pseudo
+currentDate.clone()
+```
+
+The `Date.clone()` does not take any parameters.
```
Comment (SSwiniarski) [inline content/java/concepts/date/terms/clone/clone.md:24]:
```suggestion
The `Date.clone()` method  does not take any parameters.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Rayatk I did a second review and only had one comment.

**Comment by Rayatk** [comment]
@SSwiniarski I did the changes.

**Review by SSwiniarski — APPROVED** [review]
@Rayatk looks good.

**Comment by SSwiniarski** [comment]
@Rayatk Congrats on the new entry. Live link here: https://www.codecademy.com/resources/docs/java/date/clone

image

You're currently anonymous. If you don't wish to be anonymous, check out this forum thread on how to link your GitHub and Codecademy profiles: https://discuss.codecademy.com/t/my-contribution-is-live-but-im-not-being-shown-as-a-contributor-why-is-this/758036  **Note:** it may take a few minutes for the change to register on the entry page.

**Pull request merged**